compiler/rustc_middle/src/ty/mod.rs

```diff
@@ -145,7 +146,6 @@ mod opaque_types;
 mod parameterized;
 mod rvalue_scopes;
 mod structural_impls;
-mod sty;
```

Comment:
What happens if `rustc_type_ir::sty` is made private or `pub(crate)` instead?
Is it used by other crates at all?

Same question about `pub mod select` and `pub mod util`.